modules/metastore/src/LifeCycle/LifeCycle.php

```diff
@@ -189,7 +189,7 @@ protected function distributionLoad(MetastoreItemInterface $data) {
 
     $downloadUrl = $metadata->data->downloadURL;
 
-    if (isset($downloadUrl) && !filter_var($downloadUrl, FILTER_VALIDATE_URL)) {
+    if (!empty($downloadUrl) && !filter_var($downloadUrl, FILTER_VALIDATE_URL)) {
```

Comment:
How about explicitly checking the return value of `filter_var()` for the same reason? So:
```
if (!empty($downloadUrl) && (filter_var($downloadUrl, FILTER_VALIDATE_URL) === FALSE)) {
```